app/services/curation_concerns/collections_service.rb

```diff
@@ -0,0 +1,28 @@
+module CurationConcerns
+  class CollectionsService
+    attr_reader :context
+
+    class_attribute :list_search_builder_class
+    self.list_search_builder_class = CurationConcerns::CollectionSearchBuilder
+
+    # @param [#repository,#blacklight_config,#current_ability] context
+    def initialize(context)
+      @context = context
+    end
+
+    # @param [Symbol] access :read or :edit
+    def search_results(access)
+      builder = list_search_builder(access)
+      response = context.repository.search(builder)
+      response.documents
+    end
+
+    private
+
+      def list_search_builder(access)
+        list_search_builder_class.new(context).tap do |builder|
```

Comment:
This section we would hope to replace when the upstream hydra-head blocked by browse_everything is finally unblocked.  The whole method can then just go away and be replaced by a constructor call.